### PR TITLE
TIL about Vim profiling

### PR DESCRIPTION
**Pull request by thorncp**
I finally know where to point my finger.

### CODE REVIEW

**Comment by croaky** [comment]
Very cool

File: vim/profiling.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Vim Profiling
+
+I recently encountered a situation where Vim was taking a *very* long time to
```
Comment (mike-burns) [inline vim/profiling.md]: I don't like the word "very" (Mark Twain and I have this in common). How about "excruciatingly" or "noticeably" or "annoyingly" instead?

Comment (thorncp) [inline vim/profiling.md]: I like "excruciatingly." Will update.

**Comment by JoelQ** [comment]
Very nice! How about adding a link to some documentation?

File: vim/profiling.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+time to launch. From some manual testing, I was able to isolate the problem to a
+particular directory, but I wasn't sure what was causing it. I then discovered
+Vim ships with profiling tools that would tell me how much time was spent in
+each of the functions that ran during startup.
+
+The following command will launch Vim, enable function profiling, open the
+`config/routes.rb` file, and quit, saving the profiling results to `vim.log`.
+
+```sh
+vim -c 'profile start vim.log' -c 'profile func *' -c 'e config/routes.rb' -c 'q'
```
Comment (JoelQ) [inline vim/profiling.md]: Could you add a sentence explaining what `vim -c` does?

**Comment by thorncp** [comment]
Updated with link to documentation and explanation of the the command.

**Comment by JoelQ** [comment]
Looks good! Squash, merge, and buffer at your discretion

**Pull request merged**